keras_cv/models/resnet_v1.py

```diff
@@ -0,0 +1,357 @@
+        - `max` means that global max pooling will
+            be applied.
+      num_classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+      **kwargs: Pass-through keyword arguments to `tf.keras.Model`.
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either `None` or the path to the "
+            "weights file to be loaded. Weights file not found at location: {weights}"
+        )
+
+    if include_top and not num_classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `num_classes`. "
+            f"Received: num_classes={num_classes}")
+
+    img_input = layers.Input(shape=input_shape)
+
+    if include_rescaling:
+        img_input = layers.Rescaling(1 / 255.0)(img_input)
+
+    x = layers.Conv2D(64,
+                      7,
+                      strides=2,
+                      use_bias=True,
+                      padding="same",
+                      name="conv1_conv")(img_input)
+
+    x = layers.BatchNormalization(axis=BN_AXIS,
+                                  epsilon=1.001e-5,
+                                  name="conv1_bn")(x)
+    x = layers.Activation("relu", name="conv1_relu")(x)
+
+    x = layers.MaxPooling2D(3, strides=2, padding="same", name="pool1_pool")(x)
+
+    num_stacks = len(stackwise_filters)
+
+    for stack_index in range(num_stacks):
+        x = V1Stack(
+            filters=stackwise_filters[stack_index],
+            blocks=stackwise_blocks[stack_index],
+            stride=stackwise_strides[stack_index],
+        )(x)
+
+    if include_top:
+        x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        x = layers.Dense(num_classes,
+                         activation=classifier_activation,
+                         name="predictions")(x)
+    else:
+        if pooling == "avg":
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        elif pooling == "max":
+            x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+
+    inputs = img_input
+
+    # Create model.
+    model = tf.keras.Model(inputs, x, name=name, **kwargs)
+
+    if weights is not None:
+        model.load_weights(weights)
+
+    return model
+
+
+def ResNet50(
+    include_rescaling,
+    include_top,
+    num_classes=1000,
+    weights=None,
+    input_shape=(None, None, 3),
+    pooling=None,
+    classifier_activation="softmax",
+    name="resnet50",
+    **kwargs,
+):
```

Comment:
Since this doesn't contain pre-trained weights let's have the `num_classes` and related argument as optional or required. An example is here: https://github.com/keras-team/keras-cv/pull/529/files.